Improve framework documentation

colinodell opened the pull request:
This PR makes several minor improvements to the documentation on https://pkg.go.dev/sigs.k8s.io/kustomize/kyaml/fn/framework:

- Fixes undefined variable in example code
- Adds import statements to help new users get up-and-running quicker
- Fixes the two commands needed to `run` the functions

stormqueen1990 commented:
Hi there, @colinodell!

This looks like it's in the trivial edits realm as it is right now: https://www.kubernetes.dev/docs/guide/pull-requests/#trivial-edits. Could you please take a look if there are other improvements that could be made to the same file?

Could you also please sign the CLA? We won't be able to accept your contribution until you do so. More information: https://github.com/kubernetes/community/blob/master/CLA.md

colinodell commented:
@stormqueen1990 Thanks for pointing me to those docs! I've taken this opportunity to make some other improvements to these docs, and have updated the PR title and body accordingly.  I have also signed the CLA.  I believe this is now ready for review, and I welcome any feedback you or others may have :)

stormqueen1990 reviewed (commented):
Hi there, @colinodell! 👋

This is looking good. There's just one detail that might be worth updating: in the `Example` function block, there's a `container` key missing. It currently reads:
```yaml
config.kubernetes.io/function: |
  image: image/containing/function:impl
```

whereas the format that works with `kustomize fn` is as follows:
```yaml
config.kubernetes.io/function: |
  container:
    image: image/containing/function:impl
```

Other than that, LGTM. I'll leave my `/lgtm` for now, but if you decide to make the change please feel free to ping me for a new review.

stormqueen1990 commented:
/lgtm

stormqueen1990 commented:
Hi there, @colinodell!

It seems there is a lint failure, could you please address that?
You can run the lint check locally with `make lint`.

colinodell commented:
> This is looking good. There's just one detail that might be worth updating: in the `Example` function block, there's a `container` key missing

I've added this in cc410bc23.

> It seems there is a lint failure, could you please address that?
> You can run the lint check locally with make lint.

Whoops! I've fixed that in 2252fd951. 

(I appreciate the `make lint` tip too! Unfortunately, it kept failing on some unrelated `depguard` violations in `/cmd/config`, but I figured out how to run the linter on just this one file, and it seems to be passing locally now.)

stormqueen1990 reviewed (commented):
/lgtm
/ok-to-test

koba1t commented:
Hi @colinodell 
Thanks for your contribution!

/approve
